src/Rings/mpoly-graded.jl

```diff
@@ -1399,27 +1399,27 @@ mutable struct HilbertData
   end
 end
 
-function hilbert_series(H::HilbertData, i::Int= 1)
+function hilbert_series(H::HilbertData)
   Zt, t = ZZ["t"]
   den = prod([1-t^w for w in H.weights])
-  if i==1   ### the Hilbert series with denominator prod (1-t^H.weights[i])
-    return Zt(map(ZZRingElem, H.data[1:end-1])), den
-  elseif i==2   ### the reduced Hilbert series
-    h = hilbert_series(H, 1)[1]
-    c = gcd(h, den)
-    h = divexact(h, c)
-    den = divexact(den, c)
-    return den(0)*h, den(0)*den
-  end
-  error("2nd parameter must be 1 or 2")
+  h = Zt(map(ZZRingElem, H.data[1:end-1]))
+  return h, den
+end
+
+function hilbert_series_reduced(H::HilbertData)
+  h, den = hilbert_series(H)
+  c = gcd(h, den)
```

Comment:
There is a cool function `gcd_with_cofactors` that is, for some rings, much more efficient...

Reply:
Yes, `gcd_with_cofactors` is better here.

This is unrelated to the recent discussions.